controlplane/server/internal/registry/registry_sampler.go

```diff
@@ -175,8 +181,108 @@ func (sr *SamplerRegistry) GetRegisteredInstances() []*defs.SamplerInstance {
 	return instances
 }
 
+func (sr *SamplerRegistry) createSampler(resource string, name string, tags control.Tags, capabilities control.Capabilities, config control.SamplerConfig) *defs.Sampler {
+	// Create sampler
+	sampler := defs.NewSampler(resource, name)
+	sampler.Tags = tags
+	sampler.Capabilities = capabilities
+	sampler.Config = config
+
+	return sampler
+}
+
+func (sr *SamplerRegistry) UpdateSamplerStats(resource string, name string, SamplesCollected uint64) error {
```

Comment:
It would probably be a good idea for our future selves to add a comment in here explaining that this function in addition to updating the sampler stats it also may create new sampler configurations if required.
I still think this is a bit too subtle but for now a comment may be enough to remind ourselves

Reply:
That's a good idea. Change performed as part of 7c81eda1a89dcfde0019ff20e17b0b1213a49e5b